distributed/protocol/serialize.py

```diff
@@ -86,19 +87,27 @@ def buffer_callback(f):
     return header, frames
 
 
-def pickle_loads(header, frames):
-    x, buffers = frames[0], frames[1:]
+def pickle_loads(
+    header: dict[str, Any], frames: list[bytes | bytearray | memoryview | PickleBuffer]
+) -> Any:
+    pik, buffers = frames[0], frames[1:]
 
-    writeable = header.get("writeable")
-    if not writeable:
-        writeable = len(buffers) * (None,)
+    def ensure_writeable_flag(mv: memoryview, w: bool) -> memoryview:
+        if w and mv.readonly:
+            # Can't avoid a deep copy
+            return memoryview(bytearray(mv))
```

Comment:
Adding an `assert False` to this code path makes a lot of :hankey: :hankey: :hankey: crop up to the surface.
Namely, https://github.com/dask/distributed/issues/8608